### PR TITLE
Async fixes

### LINKED ISSUE #349: test_async fails occasionally
(without valgrind)
http://bgate.mellanox.com:8888/jenkins/job/gh-ucx-pr/789/console
```
23:23:58 [ RUN      ] thread/test_async.global_event/0
23:23:58 /scrap/jenkins/jenkins/jobs/gh-ucx-pr/workspace/contrib/../test/gtest/ucs/test_async.cc:355: Failure
23:23:58 Value of: ge.count()
23:23:58   Actual: 0
23:23:58 Expected: 1
23:23:58 [  FAILED  ] thread/test_async.global_event/0, where GetParam() = 1 (8 ms)
23:23:58 [ RUN      ] thread/test_async.global_timer/0
23:23:58 [       OK ] thread/test_async.global_timer/0 (160 ms)
23:23:58 [ RUN      ] thread/test_async.max_events/0
23:23:58 [       OK ] thread/test_async.max_events/0 (8 ms)
23:23:58 [ RUN      ] thread/test_async.ctx_event/0
23:23:58 /scrap/jenkins/jenkins/jobs/gh-ucx-pr/workspace/contrib/../test/gtest/ucs/test_async.cc:406: Failure
23:23:58 Value of: le.count()
23:23:58   Actual: 0
23:23:58 Expected: 1
23:23:58 [  FAILED  ] thread/test_async.ctx_event/0, where GetParam() = 1 (6 ms)
```

### PR DESCRIPTION
**Pull request by yosefe**
Fixes #349

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com:8888/jenkins/job/gh-ucx-pr/850/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com:8888/jenkins/job/gh-ucx-pr/851/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/277/ for details. (Mellanox internal link)

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/278/ for details. (Mellanox internal link)

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/279/ for details. (Mellanox internal link)

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com:8888/jenkins/job/gh-ucx-pr/852/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/280/ for details. (Mellanox internal link)

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com:8888/jenkins/job/gh-ucx-pr/853/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com:8888/jenkins/job/gh-ucx-pr/854/ for details.

**Comment by shamisp** [comment]
autobot build

File: src/ucs/async/signal.c [hunk 1]
```diff
@@ -227,7 +227,7 @@ static void ucs_async_signal_allow(int allow)
 
     sigemptyset(&sigset);
     sigaddset(&sigset, ucs_global_opts.async_signo);
-    sigprocmask(allow ? SIG_UNBLOCK : SIG_BLOCK, &sigset, NULL);
+    pthread_sigmask(allow ? SIG_UNBLOCK : SIG_BLOCK, &sigset, NULL);
```
Comment (shamisp) [inline src/ucs/async/signal.c]: What is the goal of this change - posix compliance ?

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/281/ for details. (Mellanox internal link)

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com:8888/jenkins/job/gh-ucx-pr/855/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/282/ for details. (Mellanox internal link)

**Comment by yosefe** [comment]
@alex-mikheev  pls review

**Comment by alex-mikheev** [comment]
:+1:

**Pull request merged**